ssl/t1_lib.c

```diff
@@ -1594,6 +1594,76 @@ int ssl_setup_sigalgs(SSL_CTX *ctx)
     return ret;
 }
 
+#define SIGLEN_BUF_INCREMENT 100
+
+char *SSL_get1_builtin_sigalgs(OSSL_LIB_CTX *libctx)
+{
+    size_t i, retlen = 0, maxretlen = SIGLEN_BUF_INCREMENT;
+    const SIGALG_LOOKUP *lu;
+    EVP_PKEY *tmpkey = EVP_PKEY_new();
+    char *retval = OPENSSL_zalloc(maxretlen);
```

Comment:
zalloc seems excessive here. It is only necessary for the first byte of the buffer to be 0.

Reply:
changed in latest commit